### ISSUE TITLE
kube-proxy flags are not respected when using config file together

### ISSUE THREAD

**Issue by borgerli** [kubernetes/kubernetes#98302]
1. With following config file and start kube-proxy with flags `--config=/tmp/kube-proxy.yaml --master=https://localhost:6443 --profiling=true`. 
```
cat <<EOF > /tmp/kube-proxy.yaml
apiVersion: kubeproxy.config.k8s.io/v1alpha1
kind: KubeProxyConfiguration
clientConnection:
  kubeconfig: /var/run/kubernetes/kube-proxy.kubeconfig
hostnameOverride: 127.0.0.1
mode:
featureGates:
  AllAlpha: false
EOF
```

2. Try to access the debug endpoint, you'll get 404
```
curl 127.0.0.1:10249/debug/pprof/
404 page not found

```

Actually, not only the `profiling` flag does not take effect, all other flags which are not set in config file, are all changed to default value and then do not take effect.

**Comment by aojea** [comment]
This kube-proxy flags is always recurrent :smile: , there is a Warning to deprecate the flags
https://github.com/kubernetes/kubernetes/blob/82ebcd171909c4cd6007007fc07b025d0950153d/cmd/kube-proxy/app/server.go#L221

I think that this KEP was addressing the kube-proxy configuration problems but somehow seems stuck https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/784-kube-proxy-component-config
@thockin @andrewsykim what do you think? 
should we push more on this topic?

**Comment by borgerli** [comment]
@aojea If I understand correctly, the deprecated flags would take effect if set, and take precedence over corresponding settings in the config file. Thank you.

**Comment by jayunit100** [comment]
I also have a kep in place to clean up this stuff https://github.com/kubernetes/enhancements/pull/2158  , so that at least in the short term the ux is better @borgerli do you want to help me with that ?

**Comment by borgerli** [comment]
> I also have a kep in place to clean up this stuff [kubernetes/enhancements#2158](https://github.com/kubernetes/enhancements/pull/2158) , so that at least in the short term the ux is better @borgerli do you want to help me with that ?

Sure, glad to help.

**Comment by hanlins** [comment]
As discussed in the sig-network meeting today, we will deprecate those noop flags that got overwritten by `--config`. Discussed with other folks working on this, and our long-term goal should be promoting https://github.com/kubernetes/enhancements/pull/1439 and keep only `--config` and `--instance-config`.
/cc @knabben @dougsland @jayunit100

**Comment by hanlins** [comment]
thanks @knabben, working on this!

**Comment by jayunit100** [comment]
The above PR will get us going in the right direction here

**Comment by liggitt** [comment]
> our long-term goal should be promoting [kubernetes/enhancements#1439](https://github.com/kubernetes/enhancements/pull/1439) and keep only `--config` and `--instance-config`.

given that, does adding in flag/config merging capability make sense? warning when unused flags are specified along with config seems simpler and makes the config file able to be inspected/validated without knowing the full command line

**Comment by jayunit100** [comment]
TLDR i think this issue is frozen mostly bc the effort required to reach consensus isnt worth the gain... 
I have a   docs PR for this https://github.com/kubernetes/website/pull/28147/files#diff-cf41062ebff432932ae2a9aaa18e815626e7ad97ce7a1125cbed9a30bc8c21a6

**Comment by jayunit100** [comment]
can we close this in lieu of the above docs update ? ^
